compiler/qsc_project/src/project.rs

```diff
@@ -20,6 +20,15 @@ use thiserror::Error;
 
 pub const GITHUB_SCHEME: &str = "qsharp-github-source";
 
+#[derive(Debug, Clone, Default)]
+pub enum ProjectType {
+    #[default]
+    /// A Q# project. Described by a `qsharp.json` manifest or a single Q# file
+    QSharp,
+    /// A QASM project. Described by an `OpenQASM` source file and its includes
+    OpenQASM,
```

Comment:
I already commented about this at `load_openqasm_project`: I think you can use this enum to separate fields that are Q#-specific from QASM. Specifically , `package_graph_sources`. I don't think that type is adding much value for QASM and it looks like you're having to fight it .

`lints` is also a good candidate to make Q#-specific.